docs/rules/require-atomic-updates.md

```diff
@@ -38,32 +38,50 @@ Promise.all([getPageLength(1), getPageLength(2)]).then(pageLengths => {
 
 ## Rule Details
 
-This rule aims to report assignments to variables or properties where all of the following are true:
+This rule aims to report assignments to variables or properties in cases where the assignments may be based on outdated values.
 
-* A variable or property is reassigned to a new value which is based on its old value.
-* A `yield` or `await` expression interrupts the assignment after the old value is read, and before the new value is set.
-* The rule cannot easily verify that the assignment is safe (e.g. if an assigned variable is local and would not be readable from anywhere else while the function is paused).
+### Variables
+
+This rule reports an assignment to a variable when it detects the following execution flow in a generator or async function:
+
+1. The variable is read.
+2. A `yield` or `await` pauses the function.
+3. After the function is resumed, a value is assigned to the variable.
+
+The assignment in step 3 is reported because this flow indicates that the assignment is based on the possibly outdated value of the variable from step 1, as the variable may have been reassigned elsewhere while the function was paused in step 2.
```

Comment:
I think we can simplify this a bit:

```suggestion
The assignment in step 3 is reported because it may be incorrectly calculated because the value of the variable from step 1 may have changed between steps 2 and 3.
```

Reply:
It may also be helpful to explicitly call out the difference between local and global variables hazards with regards to this rule.

Reply:
> because it may be incorrectly calculated

I wanted to avoid the word "calculated", as that may be interpreted as if there must be a non-trivial expression that calculates the new value, and that the old value must be directly used in that expression (for example, `result = result + await something`). We had complaints that assignments such as `x = "foo"` should never be reported because there is no calculation, and the old value does not contribute to the new value. This rule reports any `read x -> await -> write x` flow, even if the write is a trivial `x = "foo"`, because the old value may have been used in a condition that determines whether or not `x = "foo"` should be executed.

> may have changed between steps 2 and 3

This could be interpreted as:

```
x; // step 1

await something; // step 2

/*
    some code between step 2 and step 3, `x` is changed here
*/

x = y; // step 3
```

I wanted to emphasize that `x` may have changed outside of this execution flow, while it was awaiting in step 2.

Reply:
Maybe “resolved” then?

Reply:
I updated this paragraph now, please take a look.